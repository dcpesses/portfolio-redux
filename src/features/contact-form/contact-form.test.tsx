/* eslint-env jest */
import { vi } from 'vitest';

import {render} from '@testing-library/react';
import ContactForm from './index';

vi.mock(('@marsidev/react-turnstile'), async(importOriginal) => {
  const actual = await importOriginal<object>();
  return {
    ...actual,
    Turnstile: vi.fn(() => {
      // Return the mock component defined above or a simple div
      return <div data-testid="mock-turnstile"></div>;
    }),
  };
});

vi.mock('react-router', () => {
  const reactRouterDom = vi.importActual('react-router');
  return {
    ...reactRouterDom,
    useNavigate: () => ({
      navigate: vi.fn()
    }),
  };
});

describe('ContactForm', () => {
  test('Should render without error', () => {
    const {container} = render(<ContactForm />);
    expect(container).toBeDefined();
  });
});
