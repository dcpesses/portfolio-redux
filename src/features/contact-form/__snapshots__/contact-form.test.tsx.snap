// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactForm > Should display alert message on caught errors > pristine 1`] = `
<form
  class=""
  data-testid="contact-form"
  id="contact-form"
  novalidate=""
>
  <div
    class="pb-3 row"
  >
    <div
      class="col-sm-6"
    >
      <label
        class="form-label"
        for="contact.Full_Name"
      >
        Full Name
      </label>
      <input
        class="form-control"
        id="contact.Full_Name"
        name="name"
        placeholder="First and Last Name"
        required=""
        type="text"
      />
      <div
        class="invalid-feedback"
      >
        Please enter your full name
      </div>
    </div>
    <div
      class="col-sm-6"
    >
      <label
        class="form-label"
        for="contact.Email_Address"
      >
        Email Address
      </label>
      <input
        class="form-control"
        id="contact.Email_Address"
        name="Email Address"
        placeholder="your@email.com"
        required=""
        type="email"
      />
      <div
        class="invalid-feedback"
      >
        Please enter a valid email address
      </div>
    </div>
  </div>
  <div
    class="pb-sm-3 row"
  >
    <div>
      <label
        class="form-label"
        for="contact.Message"
      >
        Message
      </label>
      <textarea
        class="form-control"
        id="contact.Message"
        name="message"
        placeholder="Enter your message here..."
        required=""
      />
      <div
        class="invalid-feedback"
      >
        Please enter your message.
      </div>
    </div>
  </div>
  <div
    class="mt-3 mt-sm-0 pb-2 text-center"
  >
    <div
      data-testid="mock-turnstile"
    />
  </div>
  <div
    style="opacity: 0; position: absolute; top: 0px; left: -5000px; height: 0px; width: 0px;"
  >
    <label
      for="email_subscribe_a7fdfc1ea41e_48062"
    />
    <input
      autocomplete="off"
      id="email_subscribe_a7fdfc1ea41e_48062"
      name="email_subscribe_a7fdfc1ea41e_48062"
      placeholder="Your email here"
      tabindex="-1"
      type="email"
      value=""
    />
  </div>
  <div
    class="text-center"
  >
    <button
      class="mb-2 btn btn-primary"
      type="submit"
    >
      Send Message
    </button>
  </div>
</form>
`;

exports[`ContactForm > Should submit and display alert message on form submission errors > pristine 1`] = `
<form
  class=""
  data-testid="contact-form"
  id="contact-form"
  novalidate=""
>
  <div
    class="pb-3 row"
  >
    <div
      class="col-sm-6"
    >
      <label
        class="form-label"
        for="contact.Full_Name"
      >
        Full Name
      </label>
      <input
        class="form-control"
        id="contact.Full_Name"
        name="name"
        placeholder="First and Last Name"
        required=""
        type="text"
      />
      <div
        class="invalid-feedback"
      >
        Please enter your full name
      </div>
    </div>
    <div
      class="col-sm-6"
    >
      <label
        class="form-label"
        for="contact.Email_Address"
      >
        Email Address
      </label>
      <input
        class="form-control"
        id="contact.Email_Address"
        name="Email Address"
        placeholder="your@email.com"
        required=""
        type="email"
      />
      <div
        class="invalid-feedback"
      >
        Please enter a valid email address
      </div>
    </div>
  </div>
  <div
    class="pb-sm-3 row"
  >
    <div>
      <label
        class="form-label"
        for="contact.Message"
      >
        Message
      </label>
      <textarea
        class="form-control"
        id="contact.Message"
        name="message"
        placeholder="Enter your message here..."
        required=""
      />
      <div
        class="invalid-feedback"
      >
        Please enter your message.
      </div>
    </div>
  </div>
  <div
    class="mt-3 mt-sm-0 pb-2 text-center"
  >
    <div
      data-testid="mock-turnstile"
    />
  </div>
  <div
    style="opacity: 0; position: absolute; top: 0px; left: -5000px; height: 0px; width: 0px;"
  >
    <label
      for="email_subscribe_a7fdfc1ea41e_48062"
    />
    <input
      autocomplete="off"
      id="email_subscribe_a7fdfc1ea41e_48062"
      name="email_subscribe_a7fdfc1ea41e_48062"
      placeholder="Your email here"
      tabindex="-1"
      type="email"
      value=""
    />
  </div>
  <div
    class="text-center"
  >
    <button
      class="mb-2 btn btn-primary"
      type="submit"
    >
      Send Message
    </button>
  </div>
</form>
`;

exports[`ContactForm > Should submit and display notification when form has valid data > dirty 1`] = `
<form
  class="was-validated"
  data-testid="contact-form"
  id="contact-form"
  novalidate=""
>
  <div
    class="pb-3 row"
  >
    <div
      class="col-sm-6"
    >
      <label
        class="form-label"
        for="contact.Full_Name"
      >
        Full Name
      </label>
      <input
        class="form-control"
        id="contact.Full_Name"
        name="name"
        placeholder="First and Last Name"
        required=""
        type="text"
      />
      <div
        class="invalid-feedback"
      >
        Please enter your full name
      </div>
    </div>
    <div
      class="col-sm-6"
    >
      <label
        class="form-label"
        for="contact.Email_Address"
      >
        Email Address
      </label>
      <input
        class="form-control"
        id="contact.Email_Address"
        name="Email Address"
        placeholder="your@email.com"
        required=""
        type="email"
      />
      <div
        class="invalid-feedback"
      >
        Please enter a valid email address
      </div>
    </div>
  </div>
  <div
    class="pb-sm-3 row"
  >
    <div>
      <label
        class="form-label"
        for="contact.Message"
      >
        Message
      </label>
      <textarea
        class="form-control"
        id="contact.Message"
        name="message"
        placeholder="Enter your message here..."
        required=""
      />
      <div
        class="invalid-feedback"
      >
        Please enter your message.
      </div>
    </div>
  </div>
  <div
    class="mt-3 mt-sm-0 pb-2 text-center"
  >
    <div
      data-testid="mock-turnstile"
    />
  </div>
  <div
    style="opacity: 0; position: absolute; top: 0px; left: -5000px; height: 0px; width: 0px;"
  >
    <label
      for="email_subscribe_a7fdfc1ea41e_48062"
    />
    <input
      autocomplete="off"
      id="email_subscribe_a7fdfc1ea41e_48062"
      name="email_subscribe_a7fdfc1ea41e_48062"
      placeholder="Your email here"
      tabindex="-1"
      type="email"
      value=""
    />
  </div>
  <div
    class="text-center"
  >
    <button
      class="mb-2 btn btn-primary"
      type="submit"
    >
      Send Message
    </button>
    <div
      class="form-notification"
    >
      <div
        class="fade alert alert-success alert-dismissible show"
        role="alert"
      >
        <button
          aria-label="Close alert"
          class="btn-close"
          type="button"
        />
        <div
          class="alert-heading h4"
        >
          <i
            class="bi bi-check-circle-fill"
          />
           
          Success!
        </div>
        <div
          class="alert-content"
        >
          Your message was sent successfully. Thanks for reaching out!
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`ContactForm > Should submit and display notification when form has valid data > pristine 1`] = `
<form
  class=""
  data-testid="contact-form"
  id="contact-form"
  novalidate=""
>
  <div
    class="pb-3 row"
  >
    <div
      class="col-sm-6"
    >
      <label
        class="form-label"
        for="contact.Full_Name"
      >
        Full Name
      </label>
      <input
        class="form-control"
        id="contact.Full_Name"
        name="name"
        placeholder="First and Last Name"
        required=""
        type="text"
      />
      <div
        class="invalid-feedback"
      >
        Please enter your full name
      </div>
    </div>
    <div
      class="col-sm-6"
    >
      <label
        class="form-label"
        for="contact.Email_Address"
      >
        Email Address
      </label>
      <input
        class="form-control"
        id="contact.Email_Address"
        name="Email Address"
        placeholder="your@email.com"
        required=""
        type="email"
      />
      <div
        class="invalid-feedback"
      >
        Please enter a valid email address
      </div>
    </div>
  </div>
  <div
    class="pb-sm-3 row"
  >
    <div>
      <label
        class="form-label"
        for="contact.Message"
      >
        Message
      </label>
      <textarea
        class="form-control"
        id="contact.Message"
        name="message"
        placeholder="Enter your message here..."
        required=""
      />
      <div
        class="invalid-feedback"
      >
        Please enter your message.
      </div>
    </div>
  </div>
  <div
    class="mt-3 mt-sm-0 pb-2 text-center"
  >
    <div
      data-testid="mock-turnstile"
    />
  </div>
  <div
    style="opacity: 0; position: absolute; top: 0px; left: -5000px; height: 0px; width: 0px;"
  >
    <label
      for="email_subscribe_a7fdfc1ea41e_48062"
    />
    <input
      autocomplete="off"
      id="email_subscribe_a7fdfc1ea41e_48062"
      name="email_subscribe_a7fdfc1ea41e_48062"
      placeholder="Your email here"
      tabindex="-1"
      type="email"
      value=""
    />
  </div>
  <div
    class="text-center"
  >
    <button
      class="mb-2 btn btn-primary"
      type="submit"
    >
      Send Message
    </button>
  </div>
</form>
`;
